main.rs

```diff
@@ -25,9 +25,10 @@ async fn main() {
     print_ascii_logo(); // very important
     initialise_logging_global();
 
+    let deployment_id = None;
```

Comment:
I exposed the deployment id outside of the server so it can be simply overridden by TypeDB Cloud. Plus, it's not that bad conceptually: `Server` generated its server id, but it is actually a part of a deployment, and something else should prepare the information about this deployment. Thus, if we don't have the deployment information, we are a Core server without deployment and can put out `server_id` as the `deployment_id`.